src/main/java/org/gridsuite/modification/server/modifications/BranchStatusModification.java

```diff
@@ -0,0 +1,148 @@
+import org.gridsuite.modification.server.dto.BranchStatusModificationInfos;
+
+import java.util.HashSet;
+import java.util.Objects;
+
+import static org.gridsuite.modification.server.NetworkModificationException.Type.BRANCH_NOT_FOUND;
+import static org.gridsuite.modification.server.NetworkModificationException.Type.BRANCH_ACTION_ERROR;
+
+/**
+ * @author David Braquart <david.braquart at rte-france.com>
+ */
+public class BranchStatusModification extends AbstractModification {
+
+    private final BranchStatusModificationInfos modificationInfos;
+
+    public BranchStatusModification(BranchStatusModificationInfos modificationInfos) {
+        this.modificationInfos = modificationInfos;
+    }
+
+    @Override
+    public void apply(Network network, Reporter subReporter) {
+        String branchId = modificationInfos.getEquipmentId();
+        Branch<?> branch = network.getBranch(branchId);
+        if (branch == null) {
+            throw new NetworkModificationException(BRANCH_NOT_FOUND, branchId);
+        }
+
+        String branchType = branch.getType() == IdentifiableType.LINE ? "Line" : "2WTransformer";
+        switch (modificationInfos.getAction()) {
+            case LOCKOUT:
+                applyLockoutBranch(subReporter, branch, branchType);
+                break;
+            case TRIP:
+                applyTripBranch(subReporter, branch, branchType, network);
+                break;
+            case SWITCH_ON:
+                applySwitchOnBranch(subReporter, branch, branchType);
+                break;
+            case ENERGISE_END_ONE:
+                applyEnergiseBranchEnd(subReporter, branch, branchType, Branch.Side.ONE);
+                break;
+            case ENERGISE_END_TWO:
+                applyEnergiseBranchEnd(subReporter, branch,branchType, Branch.Side.TWO);
+                break;
+            default:
+                throw NetworkModificationException.createBranchActionTypeUnsupported(modificationInfos.getAction());
+        }
+    }
+
+    private void applyLockoutBranch(Reporter subReporter, Branch<?> branch, String branchType) {
+        if (disconnectAllTerminals(branch)) {
+            branch.newExtension(BranchStatusAdder.class).withStatus(BranchStatus.Status.PLANNED_OUTAGE).add();
+        } else {
+            throw new NetworkModificationException(BRANCH_ACTION_ERROR, "Unable to disconnect all branch ends");
+        }
+        subReporter.report(Report.builder()
+            .withKey("lockout" + branchType + "Applied")
+            .withDefaultMessage(branchType + " ${id} (id) : lockout applied")
+            .withValue("id", branch.getId())
+            .withSeverity(TypedValue.INFO_SEVERITY)
+            .build());
+    }
+
+    private void applyTripBranch(Reporter subReporter, Branch<?> branch, String branchType, Network network) {
+        var trip = new BranchTripping(branch.getId());
+        var switchToDisconnect = new HashSet<Switch>();
```

Comment:
A mark  of plural (-> `switches` I guess) would be welcome

Reply:
its just a move of existing code, but you're right. I will change